cmd/provisioner-localpv/app/config.go

```diff
@@ -0,0 +1,217 @@
+	if len(strings.TrimSpace(stgType)) == 0 {
+		return "hostpath"
+	}
+	return stgType
+}
+
+//GetPath returns a valid PV path based on the configuration
+// or an error. The Path is constructed using the following rules:
+// If AbsolutePath is specified return it. (Future)
+// If PVPath is specified, suffix it with BasePath and return it. (Future)
+// If neither of above are specified, suffix the PVName to BasePath
+//  and return it
+// Also before returning the path, validate that path is safe
+//  and matches the filters specified in StorageClass.
+func (c *VolumeConfig) GetPath() (string, error) {
+	//This feature need to be supported with some more
+	// security checks are in place, so that rouge pods
+	// don't get access to node directories.
+	//absolutePath := c.getValue(KeyPVAbsolutePath)
+	//if len(strings.TrimSpace(absolutePath)) != 0 {
+	//	return c.validatePath(absolutePath)
+	//}
+
+	basePath := c.getValue(KeyPVBasePath)
+	if strings.TrimSpace(basePath) == "" {
+		return "", errors.Errorf("failed to get path: base path is empty")
+	}
+
+	//This feature need to be supported after the
+	// security checks are in place.
+	//pvRelPath := c.getValue(KeyPVRelativePath)
+	//if len(strings.TrimSpace(pvRelPath)) == 0 {
+	//	pvRelPath = c.pvName
+	//}
+
+	pvRelPath := c.pvName
+	path := filepath.Join(basePath, pvRelPath)
+
+	return c.validatePath(path)
+}
+
+//getValue is a utility function to extract the value
+// of the `key` from the ConfigMap object - which is
+// map[string]interface{map[string][string]}
+// Example:
+// {
+//     key1: {
+//             value: value1
+//             enabled: true
+//           }
+// }
+// In the above example, if `key1` is passed as input,
+//   `value1` will be returned.
+func (c *VolumeConfig) getValue(key string) string {
+	if configObj, ok := util.GetNestedField(c.options, key).(map[string]string); ok {
+		if val, p := configObj[string(mconfig.ValuePTP)]; p {
+			return val
+		}
+	}
+	return ""
+}
+
+//validatePath checks for the sanity of the PV path
+func (c *VolumeConfig) validatePath(path string) (string, error) {
+	//Validate that path is well formed.
+	path, err := filepath.Abs(path)
+	if err != nil {
+		return "", err
+	}
+
+	//Validate that root directories are not used as PV paths
+	//Valid path should have a basepath (which is not /) and
+	// a subpath for the PV.
+	path = strings.TrimSuffix(path, "/")
+	parentDir, volumeDir := c.extractSubPath(path)
+	//parentDir = strings.TrimSuffix(parentDir, "/")
+	//volumeDir = strings.TrimSuffix(volumeDir, "/")
+	if parentDir == "" || volumeDir == "" {
+		// it covers the `/` case
+		return "", errors.Errorf("validate the volume path: {%v} is invalid: cannot find parent dir or volume dir", path)
+	}
+
+	//TODO: Validate against blacklist or whitelist of paths
+	return path, nil
+}
+
+//extractSubPath is utility function to split directory from path
+func (c *VolumeConfig) extractSubPath(path string) (string, string) {
```

Comment:
Object `c` is not used.

Reply:
intended for future use with respect to validations stored in volumeconfig.